src/commands/local.ts

```diff
@@ -171,18 +173,89 @@ export async function prepareLocalBackendEnvironment(
     }
 }
 
-// Function that starts the local environment.
-// It also monitors for changes in the user's code and restarts the environment when changes are detected.
+// Function that starts the local environment. It starts the backend watcher and the frontends.
 export async function startLocalEnvironment(options: GenezioLocalOptions) {
     await GenezioTelemetry.sendEvent({
         eventType: TelemetryEventTypes.GENEZIO_LOCAL,
         commandOptions: JSON.stringify(options),
     });
     const yamlConfigIOController = new YamlConfigurationIOController(options.config);
     const yamlProjectConfiguration = await yamlConfigIOController.read();
-    const backendConfiguration = yamlProjectConfiguration.backend;
+    // This mutex is used to make the frontends wait until the first Genezio SDK is generated.
+    const sdkSynchronizer = new Mutex();
+    // It is locked until the first Genezio SDK is generated.
+    sdkSynchronizer.acquire();
+
+    if (!yamlProjectConfiguration.backend && !yamlProjectConfiguration.frontend) {
+        throw new UserError(
+            "No backend or frontend components found in the genezio.yaml file. You need at least one component to start the local environment.",
+        );
+    } else if (
+        yamlProjectConfiguration.frontend &&
+        yamlProjectConfiguration.frontend.every((f) => !f.scripts?.start)
+    ) {
+        throw new UserError(
```

Comment:
Do we want this? I feel like I should be able to choose to opt out of this feature if I don't want this or would like to use more terminals

Reply:
This error happens only if you don't have a backend. If you run local only with frontends and no frontend has a start script, i think this should be an error, otherwise the program will just instantly exit.

Reply:
You are right!